In [2]:
import pandas as pd

df = pd.read_csv('이원분산분석_실습.csv', sep=',')
df

store  area  satisfaction
0       1     1             1
1       1     2             5
2       1     3             4
3       1     1             2
4       1     2             3
5       1     3             3
6       1     1             2
7       1     2             4
8       1     3             4
9       2     1             3
10      2     2             5
11      2     3             4
12      2     1             3
13      2     2             5
14      2     3             4
15      2     1             2
16      2     2             4
17      2     3             4
18      3     1             1
19      3     2             3
20      3     3             3
21      3     1             1
22      3     2             2
23      3     3             3
24      3     1             2
25      3     2             3
26      3     3             3
27      3     1             2
28      3     2             3
29      3     3             4

In [3]:
!pip install pingouin #For ANOVA analysis

   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 204.4/204.4 kB 8.6 MB/s eta 0:00:00


In [5]:
!pip install scikit-posthocs #Post-hoc analysis after Two-way ANOVA

In [6]:
import pingouin as pg
import scikit_posthocs

df1 = df[['store', 'area', 'satisfaction']]
print(pg.anova(dv='satisfaction', between=['store', 'area'], data=df1))

         Source         SS    DF         MS          F         p-unc       np2
0         store   8.422222   2.0   4.211111  12.339535  2.859377e-04  0.540271
1          area  20.466667   2.0  10.233333  29.986047  7.011021e-07  0.740651
2  store * area   1.811111   4.0   0.452778   1.326744  2.927256e-01  0.201733
3      Residual   7.166667  21.0   0.341270        NaN           NaN       NaN


In this two-way ANOVA analysis, we examined whether two factors—store and area—have an effect on satisfaction, and whether there is an interaction effect between them. Here are the results explained in simple terms:

Effect of store:
The statistical test shows an F-value of 12.34 with a p-value of 0.00029.
Since this p-value is less than 0.05 (written as p < 0.05), the result is considered statistically significant.
-> This means that satisfaction levels differ significantly depending on the store.

Effect of area:
The F-value is 29.99 and the p-value is 0.0000007011, which is also much smaller than 0.05.
-> This indicates that area has a strong and statistically significant effect on satisfaction.

Interaction between store and area:
The F-value here is 1.33 and the p-value is 0.2927, which is greater than 0.05 (written as p > 0.05).
-> This result is not statistically significant, meaning that the effect of store on satisfaction does not depend on the area, and vice versa.



In [7]:
df1['store'].astype(str)
print(scikit_posthocs.posthoc_scheffe(df1, val_col='satisfaction', group_col='store'))

          1         2         3
1  1.000000  0.411815  0.426107
2  0.411815  1.000000  0.033807
3  0.426107  0.033807  1.000000


2,3번 브랜드의 만족도에 차이가 있다.

In [ ]:
df1['area'].astype(str)
print(scikit_posthocs.posthoc_scheffe(df1, val_col='satisfaction', group_col='area'))

          1         2         3
1  1.000000  0.000139  0.000285
2  0.000139  1.000000  0.962003
3  0.000285  0.962003  1.000000


In [ ]:
일원 분산 분석은 하나의 요인과 하나의 변수.  하나의 요인으로 이 변수에 차이가 있는지 없는지를 검증.
2개의 요인가 이 하나의 변수에 영향을 미치는지를 분석하는 것.
멀티는 manova검색
1,2 유의미
1,3 유의미